slice/internal/controller/workload_controller.go

```diff
@@ -64,34 +85,74 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	log.V(2).Info("Reconcile Workload")
 
 	if r.shouldFinalize(wl) {
-		if controllerutil.ContainsFinalizer(wl, CleanupSliceFinalizerName) {
+		if controllerutil.ContainsFinalizer(wl, SliceControllerName) {
 			err = r.client.Delete(ctx, r.newEmptySlice(wl))
 			if client.IgnoreNotFound(err) != nil {
 				return ctrl.Result{}, err
 			}
-			controllerutil.RemoveFinalizer(wl, CleanupSliceFinalizerName)
+			controllerutil.RemoveFinalizer(wl, SliceControllerName)
 			if err := r.client.Update(ctx, wl); err != nil {
-				return ctrl.Result{}, err
+				if !apierrors.IsNotFound(err) {
+					log.Error(err, "Failed to remove finalizer")
+				}
+				return ctrl.Result{}, client.IgnoreNotFound(err)
 			}
 			log.V(5).Info("Removed finalizer")
 		}
 		return ctrl.Result{}, nil
 	}
 
-	if controllerutil.AddFinalizer(wl, CleanupSliceFinalizerName) {
-		if err := r.client.Update(ctx, wl); err != nil {
-			log.V(5).Info("Added finalizer")
-			return ctrl.Result{}, err
+	ac, err := r.sliceAC(ctx, wl)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if ac == nil {
+		log.V(5).Info("Admission check not found – ignoring reconciliation for now")
+		return reconcile.Result{}, nil
+	}
```

Comment:
Which controller is responsible for creating the admission check for this workload? I think this could be done here when it is not found